isis/tests/FunctionalTestsCkwriter.cpp

```diff
@@ -0,0 +1,213 @@
+  targetPosition += kernels["TargetPosition"][2];
+  kernels.addKeyword(targetPosition, PvlContainer::InsertMode::Replace);
+
+  PvlKeyword instrumentPointing("InstrumentPointing");
+  instrumentPointing += options.GetFileName("TO");
+  instrumentPointing += "$viking1/kernels/fk/vo1_v10.tf";
+  kernels.addKeyword(instrumentPointing, PvlContainer::InsertMode::Replace);
+
+  PvlKeyword instrumentPosition("InstrumentPosition");
+  instrumentPosition += kernels["InstrumentPosition"][1];
+  kernels.addKeyword(instrumentPosition, PvlContainer::InsertMode::Replace);
+
+  newKernelCube.reopen("rw");
+
+  Camera *newCamera = nullptr;
+
+  try {
+    newCamera = newKernelCube.camera();
+  } catch(IException &e) {
+    FAIL() << "Unable to generate camera with new ck kernel: " << e.toString().toStdString().c_str() << std::endl;
+  }
+
+  SpiceRotation *newKernelRotation = newCamera->instrumentRotation();
+  SpiceRotation *originalRotation = testCube->camera()->instrumentRotation();
+
+  Table instPointingTable("InstrumentPointing");
+  testCube->read(instPointingTable);
+  double startTime = double(instPointingTable.Label()["CkTableStartTime"]);
+
+  newKernelRotation->SetEphemerisTime(startTime);
+  originalRotation->SetEphemerisTime(startTime);
+
+  ASSERT_EQ(newKernelRotation->cacheSize(), originalRotation->cacheSize());
+
+  for (int i = 0; i < newKernelRotation->TimeBasedMatrix().size(); i++) {
+    ASSERT_DOUBLE_EQ(newKernelRotation->TimeBasedMatrix()[i],
+                     originalRotation->TimeBasedMatrix()[i]);
+  }
+
+  for (int i = 0; i < newKernelRotation->AngularVelocity().size(); i++) {
+    ASSERT_DOUBLE_EQ(newKernelRotation->AngularVelocity()[i],
+                     originalRotation->AngularVelocity()[i]);
+  }
+}
+
+TEST_F(DefaultCube, FunctionalTestsCkwriterFromlist) {
+  Pvl appLog;
+  FileList cubeList;
+  cubeList.append(testCube->fileName());
+
+  QString cubeListFile = tempDir.path() + "/cubes.lis";
+  cubeList.write(cubeListFile);
+  QVector<QString> args = {"fromlist=" + cubeListFile,
+                           "to=" + tempDir.path() + "/newKernel.bc"};
+
+  UserInterface options(APP_XML, args);
+  try {
+   ckwriter(options, &appLog);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to write kernel file: " << e.toString().toStdString().c_str() << std::endl;
+  }
+
+  Cube newKernelCube;
+  newKernelCube.fromLabel(tempDir.path() + "/newKernelCube.cub", label, "rw");
+
+  PvlGroup &kernels = newKernelCube.label()->findObject("IsisCube").findGroup("Kernels");
+  PvlKeyword targetPosition("TargetPosition");
+  targetPosition += kernels["TargetPosition"][1];
+  targetPosition += kernels["TargetPosition"][2];
+  kernels.addKeyword(targetPosition, PvlContainer::InsertMode::Replace);
+
+  PvlKeyword instrumentPointing("InstrumentPointing");
+  instrumentPointing += options.GetFileName("TO");
+  instrumentPointing += kernels["InstrumentPointing"][2];
+  kernels.addKeyword(instrumentPointing, PvlContainer::InsertMode::Replace);
```

Comment:
An alternative is to get the keyword as a reference and simply modify it in place.